src/App.tsx

```diff
@@ -1,61 +1,132 @@
 import './App.scss';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+import { TodoList } from './components/TodoList';
+import React, { useState } from 'react';
+import { User } from './types/User';
+import { Todos } from './types/Todos';
 
 export const App = () => {
+  const getUserById = (userId: number): User | null =>
+    usersFromServer.find(user => user.id === userId) || null;
+
+  const initialTodos: Todos[] = todosFromServer.map(todo => ({
+    ...todo,
+    user: getUserById(todo.userId),
+  }));
+
+  const [userList] = useState(usersFromServer);
+  const [todosList, setTodosList] = useState(initialTodos);
+
+  const [titleInput, setTitleInput] = useState('');
+  const [userSelected, setUserSelected] = useState('');
+
+  const [titleInputError, setTitleInputError] = useState(false);
+  const [userSelectError, setUserSelectError] = useState(false);
+
+  const receiveTheMaxId = () => {
+    const todoIds = todosList.map(todo => todo.id);
+
+    return Math.max(...todoIds) + 1;
+  };
+
+  const createNewTodo = (): Todos => ({
+    id: receiveTheMaxId(),
+    title: titleInput.trim(),
+    completed: false,
+    userId: +userSelected,
+    user: getUserById(+userSelected),
```

Comment:
The `getUserById` function can return `null`, which would result in a `null` user object in the new todo. Consider adding a check to handle this scenario to prevent potential errors.